host/manager_test.go

```diff
@@ -19,7 +19,10 @@ func validateManager(m Manager, expectedHostname string, t *testing.T) {
 }
 
 func TestNewManager(t *testing.T) {
-	m, err := NewManager(false)
+	// we should save fqdn for NewManager since os.Hostname will return /proc/sys/kernel/hostname and it can be fqdn
+	// underhood NewManager will use same call, but if we don't save fqdn we are trying to split that name
+	// In that can this test will fail anyway
+	m, err := NewManager(true)
```

Comment:
`os.Hostname()` calls `sysctl kern.hostname` - It only corresponds to the contents of `/proc/sys/kernel/hostname` on some Linux systems and may or may not do so in the future. 

I agree that the test should pass, but I'm a bit worried that this is changing the semantics of the test. This is now not testing the splitting anymore. Should the expected host name be different instead?

Reply:
I've rewritten test to make check more predictable:)